src/lib/router.ts

```diff
@@ -30,4 +32,61 @@ router.post('/matches', async (req, res, next) => {
   }
 });
 
+router.post('/matches/:id/highlights', async (req, res, next) => {
+  const id = req.params.id;
+  const { timestamp, type, videoSrc } = req.body;
+
+  if (!isValid(id)) {
+    res.status(400).send('Invalid match id');
+    return;
+  }
+
+  if (
+    typeof timestamp !== 'number' ||
+    typeof type !== 'string' ||
+    typeof videoSrc !== 'string'
+  ) {
+    res.status(400).send('Invalid payload');
+    return;
+  }
+
+  const newHighlight: MatchHighlight = {
+    timestamp,
+    type,
+    videoSrc,
+  };
+
+  try {
+    const result = await getMatchesCollection().findOneAndUpdate(
+      { _id: new ObjectId(id) },
+      { $push: { highlights: newHighlight } }
+    );
+
+    if (!result) {
+      res.status(404).send("didn't found match to update");
+      return;
+    }
+
+    res.status(200).json(result);
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.get('/matches', async (_req, res, next) => {
+  try {
+    const data = await getMatchesCollection().find({});
+    const result = await data.toArray();
+
+    if (!result.length) {
+      res.status(404).send('no matches found');
+      return;
+    }
```

Comment:
404 is a client side error code and tells the client, that a resource was requested which doesn't exist.
But the client didn't do anything wrong here.
Actually, an empty array is not an error. It's fine that no matches exist.
Imagine that you add a search query like `/api/matches?gameId=9999`.

Reply:
true, if no matches are in there, there is nothing wrong ;-)